homeassistant/components/hue.py

```diff
@@ -0,0 +1,237 @@
+"""
+homeassistant.components.hue
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Philips Hue bridge.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/hue/
+"""
+
+import logging
+import socket
+from datetime import timedelta
+from urllib.parse import urlparse
+
+from homeassistant import bootstrap
+from homeassistant.util import Throttle
+from homeassistant.loader import get_component
+from homeassistant.components import discovery
+from homeassistant.const import (
+    CONF_HOST, ATTR_SERVICE, ATTR_DISCOVERED, EVENT_PLATFORM_DISCOVERED)
+
+DOMAIN = 'hue'
+REQUIREMENTS = ['phue==0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+MIN_TIME_BETWEEN_FORCED_UPDATES = timedelta(milliseconds=100)
+
+PHUE_CONFIG_FILE = "phue.conf"
+
+DISCOVER_LIGHTS = 'hue.light'
+DISCOVER_SWITCHES = 'hue.switch'
+DISCOVER_SENSORS = 'hue.sensor'
+
+# terminology used by phue
+ATTR_HUE_LIGHTS = 'lights'
+ATTR_HUE_BRIDGEID = 'bridgeid'
+ATTR_HUE_BRIDGEIP = 'ipaddress'
+
+# Preparation for tap, switches and dimmers
+DISCOVERY_COMPONENTS = [
+    ('light', DISCOVER_LIGHTS, ATTR_HUE_LIGHTS,),
+    ('switch', DISCOVER_SWITCHES, '',),
+    ('sensor', DISCOVER_SENSORS, '',),
+]
+
+HUEBRIDGE = None
+
+# Map ip to request id for configuring
+_CONFIGURING = {}
+
+
+def setup(hass, config):
+    """
+    Setup a Hue bridge
+
+    Set up a 'global' to contain bridge instances to be called.
+    If the bridge is non-local, it cannot be discovered. Multiple bridge can be
+    listed in configuration.
+    Setup discovery
+    """
+
+    # This dict will hold all bridges. The dict key is the bridgeid.
+    # It's much easier to just use 'host', but deals better with multiple IPs
+    # or DHCP swtiching bridge ip?
+    global HUEBRIDGE
+    HUEBRIDGE = {}
+
+    # Fetch config
+    conf = config.get(DOMAIN, {})
+    if not isinstance(conf, list):
+        conf = [conf]
+    for entry in conf:
+        host = entry.get(CONF_HOST, None)
+        if not host:
+            continue
+        # FIXME / This can only happen when discovery beats configuration
+        # loading right?
+        if host in _CONFIGURING:
+            continue
+
+        _LOGGER.info('Attempting bridge setup for %s', host)
+        setup_huebridge(hass, host)
+
+    comp_name = 'light'
+    component = get_component(comp_name)
+    bootstrap.setup_component(hass, component.DOMAIN, config)
```

Comment:
Yes, this is needed. On line 231, you fire EVENT_PLATFORM_DISCOVERED for each light you detect on the Hub. This event will get routed to the *light* component without going through *discovery*. Also, you will need to set this component to be the [service handler in Discovery](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/discovery.py#L34)